pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -916,3 +918,103 @@ func MakeNextPodFunc(queue SchedulingQueue) func() *v1.Pod {
 func podInfoKeyFunc(obj interface{}) (string, error) {
 	return cache.MetaNamespaceKeyFunc(obj.(*podInfo).pod)
 }
+
+// PodBackoffMap is a structure that stores backoff related information for pods
+type PodBackoffMap struct {
+	// lock for performing actions on this PodBackoffMap
+	lock sync.Mutex
+	// initial backoff duration
+	initialDuration time.Duration
+	// maximal backoff duration
+	maxDuration time.Duration
+	// map for pod -> number of attempts for this pod
+	podAttempts map[ktypes.NamespacedName]int
+	// map for pod -> lastUpdateTime pod of this pod
+	podLastUpdateTime map[ktypes.NamespacedName]time.Time
```

Comment:
If we used PodBackoffMap only for the PriorityQueue, we wouldn't need this here. `podInfo` already has a timestamp that can be used. For now, we can have it here until we remove the FIFO queue in the next step.

Reply:
Yes, you are right. `podInfo.timestamp` can represent the time pod starts backoff in `AddUnschedulableIfNotPresent `.  But the comment is not needed to change. Because  timestamp represents the time pod added to the scheduling queue(no matter activeQ、podBackoffQ、unschedulableQ)

Reply:
> shall we change this PR to use podInfo here?

We cannot do it in this PR. The FIFO scheduling queue does not use `podInfo`. The changes you made in the `factory.go` apply to FIFO queue only.
In a follow-up PR, we can remove the FIFO queue and switch to `podInfo` and delete this `podLastUpdateTime` map.